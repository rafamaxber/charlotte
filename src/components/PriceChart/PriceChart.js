import React from 'react';
import styled from 'styled-components';
import BackButtonImage from '../../assets/back-description.svg';

const WrapperChart = styled.div`
  width: 600px;
`;
const StageChart = styled.div`
  width: 500px;
`;
const Title = styled.div`
  font-family: 'Heebo', sans-serif; 
  font-style: normal;
  font-weight: bold;
  line-height: 26px;
  font-size: 16px;
  letter-spacing: 1.81818px;
  display: inline-block;
  width: 49%;
  text-transform: uppercase;
  color: #F98100;
  vertical-align: middle;
`;
const BackButton = styled.div`
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  text-align: right;
  width: 49%;
  font-family: 'Heebo', sans-serif; 
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 12px;
  letter-spacing: 1.36364px;
  color: #555555;
`;
const BackButtonIcon = styled.img.attrs({
  src: BackButtonImage
})`
  width: 26px;
  display: inline-block;
  vertical-align: top;
  margin-right: 10px;
`;
const Chart = styled.div`
  border-bottom: 2px solid #d9d9d9;
  min-height: 180px;
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
`;
const Bar = styled.div`
  width: 27px;
  height: ${props => props.size}px;
  position: relative;
  bottom: 0;
  transition: .3s ease;
  background: linear-gradient(0deg, rgba(238, 111, 72, 0.6) 0%, rgba(249, 129, 0, 0.6) 100%);
  &:hover,
  &:focus,
  &:active {
    transition: .3s ease;
    background: linear-gradient(0deg, #EE6F48 0%, #F98100 100%);
  }
  &:hover .chart-tooltip {
    visibility: visible!important;
    opacity: 1!important;
  }
`;
const Tooltip = styled.div`
  font-family: 'Heebo', sans-serif; 
  visibility: hidden;
  font-size: 15px;
  position: absolute;
  background-color: #484848;
  color: #fff;
  text-align: center;
  padding: 5px 10px;
  border-radius: 6px;
  z-index: 1;
  width: 70px;
  top: -35px;
  left: -21px;
  opacity: 0;
  transition: opacity .6s;
  &:after {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: #555 transparent transparent transparent;
  }
`;
const ChartLabel = styled.div`
  font-family: 'Heebo', sans-serif; 
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 12.8px;
  text-align: center;
  letter-spacing: 1.45455px;
  color: #B5B5B5;
  bottom: -25px;
  position: absolute;
`;

export default ({ history, closeChart }) => (
  <WrapperChart>
    <StageChart>
      <Title>Price history for 2017</Title>
      <BackButton onClick={closeChart}>
        <BackButtonIcon></BackButtonIcon>
        Back to description
      </BackButton>
      <Chart>
        {
          history.map(item => (
            <Bar size={item.value / 5} key={item.month}>
              <Tooltip className="chart-tooltip">${item.value}</Tooltip>
              <ChartLabel>{item.month}</ChartLabel>
            </Bar>
          ))
        }
      </Chart>
    </StageChart>
  </WrapperChart>
);
