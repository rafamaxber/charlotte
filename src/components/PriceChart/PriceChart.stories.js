import React from 'react';
import { storiesOf } from '@storybook/react';
import PriceChart from './PriceChart';

const mock = [{ "price_history": [{ "month": "Jan", "value": 438 }, { "month": "Feb", "value": 400.1 }, { "month": "Mar", "value": 425 }, { "month": "Apr", "value": 433.6 }, { "month": "May", "value": 428.4 }, { "month": "Jun", "value": 415 }, { "month": "Jul", "value": 412 }] }];

storiesOf('PriceChart', module)
  .add('PriceChart', () => (
    <PriceChart
      history={mock[0].price_history}
    ></PriceChart>
  ));
