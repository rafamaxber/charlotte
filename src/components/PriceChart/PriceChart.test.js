import React from 'react';
import renderer from 'react-test-renderer';
import PriceChart from './PriceChart';

const mock = [{ "price_history": [{ "month": "Jan", "value": 438 }, { "month": "Feb", "value": 400.1 }, { "month": "Mar", "value": 425 }, { "month": "Apr", "value": 433.6 }, { "month": "May", "value": 428.4 }, { "month": "Jun", "value": 415 }, { "month": "Jul", "value": 412 }] }];

describe('PriceChart', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <PriceChart
        history={mock[0].price_history}
      ></PriceChart>
    );
    expect(component).toMatchSnapshot();
  });
});
