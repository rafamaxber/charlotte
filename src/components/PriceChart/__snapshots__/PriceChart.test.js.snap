// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceChart Should match with snapshot 1`] = `
<div
  className="sc-bdVaJa eQIsiy"
>
  <div
    className="sc-bwzfXH rYvhI"
  >
    <div
      className="sc-htpNat lnWDRr"
    >
      Price history for 2017
    </div>
    <div
      className="sc-bxivhb idpqiB"
      onClick={undefined}
    >
      <img
        className="sc-ifAKCX fTxmoU"
        src="back-description.svg"
      />
      Back to description
    </div>
    <div
      className="sc-EHOje fvkpsD"
    >
      <div
        className="sc-bZQynM gziNKJ"
        size={87.6}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          438
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Jan
        </div>
      </div>
      <div
        className="sc-bZQynM Sgapf"
        size={80.02000000000001}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          400.1
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Feb
        </div>
      </div>
      <div
        className="sc-bZQynM biprHx"
        size={85}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          425
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Mar
        </div>
      </div>
      <div
        className="sc-bZQynM kCXycv"
        size={86.72}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          433.6
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Apr
        </div>
      </div>
      <div
        className="sc-bZQynM lnmBcL"
        size={85.67999999999999}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          428.4
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          May
        </div>
      </div>
      <div
        className="sc-bZQynM iTEwDt"
        size={83}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          415
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Jun
        </div>
      </div>
      <div
        className="sc-bZQynM hRFoqL"
        size={82.4}
      >
        <div
          className="chart-tooltip sc-gzVnrw hYwrRH"
        >
          $
          412
        </div>
        <div
          className="sc-htoDjs cVtJOz"
        >
          Jul
        </div>
      </div>
    </div>
  </div>
</div>
`;
